plugin/completion/flags_manager.py

```diff
@@ -298,10 +299,9 @@ def write_flags_to_file(self, new_flags, file_path, strategy):
             if flag_strategy == "keep_old":
                 return
             if flag_strategy == "merge":
-                # union of two flags sets
-                curr_flags = set(self.flags_from_clang_file(
-                    File(file_path)))
-                new_flags = new_flags.union(curr_flags)
+                # union of two flag arrays
+                curr_flags = self.flags_from_clang_file(File(file_path))
+                new_flags = curr_flags + list(set(new_flags) - set(curr_flags))
```

Comment:
@rchl actually after looking at this line again, it is better than having just one set for the order of flags, but even this does not guarantee that their order will be preserved, as `set(new_flags) - set(curr_flags)` will put them in arbitrary order.